commands/new_function.go

```diff
@@ -145,3 +164,73 @@ func validTemplate(lang string) bool {
 
 	return found
 }
+
+// insertAtRegexpMatch takes a full YAML file data (fileData), and
+// data to insert at the match location (insertData). Returns a string
+// that is fileData with functionYAMLData inserted in the 'functions:'
+// block of the YAML file.
+func insertFunctionIntoYAML(fileData string, functionYAMLData string) string {
```

Comment:
What is this doing @ericstoekl ?

Reply:
We need to append the function data to the existing YAML, but keep the YAML in its original order. I do this by grabbing the top-level objects in the YAML (for now that means just `provider:` and `functions:`, i.e. the lines of the YAML file that do not begin with whitespace and are not comments) and then inserting the new data into the `functions:` section. If the `functions:` section comes before the `provider:` section, then we need to know where the `provider:` section is.

The other way of doing this that does not use a regex is to loop through every line of the file, scanning for top-level objects (`functions:` and `provider:`). I figured the regex would be less code and therefore better, but I can see why we might not want to use regex. Do you think this is a reasonable situation for regex, or would looping through each line of the file be better? Or is there a better solution that I haven't thought of?

Reply:
@ericstoekl can you remind me what the need to keep the `yaml` in the same order was for please? I've forgotten the context here..